bin/meta_xform.py

```diff
@@ -0,0 +1,180 @@
+        else:
+            rec[k] = dict(zip(LANG_KEYS, (
+                rec.pop(k, None),
+                rec.pop(k_fra, None))))
+
+    # convert subject english-sp-sp-french content to fluent text
+    rec['subject'] = [
+        sd_new_dfc['subject'][s.lstrip().split(SP_SP, 1)[0]]
+            for s in rec['subject']]
+
+    # convert frenquency english-sp-pipe-sp-french content to fluent text
+    freq = rec.pop('maintenance_and_update_frequency')
+    if (freq):
+        rec['frequency'] = sd_new_dfc[u'frequency'][
+            freq.lstrip().split(SP_PIPE_SP, 1)[0]]
+
+    # convert geo region english-sp-sp-french content to fluent text
+    rec['geographic_region'] = [
+        sd_new_dfc[u'geographic_region'][gr.lstrip().split(SP_SP, 1)[0]]
+            for gr in rec['geographic_region']]
+
+    # merge per-resource name, name_fra to fluent text
+    for r in rec['resources']:
+        r['name'] = dict(
+            zip(LANG_KEYS, (r.pop('name', None), r.pop('name_fra', None))))
+
+    _process.count[1] += 1
+    logging.debug('Skipped {0}, processed {1}'.format(*_process.count))
+    logging.debug(simplejson.dumps(rec, indent=4 * ' '))
+    return rec
+_process.count = [0, 0]
+
+def _main(fpath_jsonl_old, fpath_jsonl_new):
+    """
+    With input JSONL data file, open and process a line at a time;
+    write output to gzipped JSONL new-style file
+    """
+    logging.basicConfig(format='%(levelname)s:%(message)s', level=logging.INFO)
+
+    with gzip.open(fpath_jsonl_old, 'rb') as fp_in:
+        with gzip.open(fpath_jsonl_new, 'wb') as fp_out:
+            try:
+                for line in fp_in:
+                    rec = _process(line.rstrip())
+                    if (rec):
+                        fp_out.write(simplejson.dumps(rec) + '\n')
+            except IOError:
+                print >> sys.stderr, (
+                    'Error: input file <{0}> not gzipped'.format(
+                        fpath_jsonl_old))
+
+def usage():
+    """
+    Display usage message
+    """
+    print >> sys.stderr, (
+        'Usage: python {0} <in-file> <out-file>'.format(sys.argv[0]))
+    print >> sys.stderr, (
+        'where <in-file> is a gzipped ckanext-canada@v2.1 metadata file')
+    print >> sys.stderr, (
+        'and <out-file> will be a gzipped ckanext-canada@v2.3 metadata file.')
+    sys.exit(-1)
+
+def _set_new_schema_dataset_choices():
+    """
+    Initialize tree of choices (label:value) for new schema dataset
+    choice fields
+    """
+    global sd_new_dfc
+    _HERE = path.dirname(path.abspath(__file__))
+    _SCHEMAS_DIR = path.join(path.dirname(_HERE),'ckanext/canada/schemas')
+    _JSON_NAME = path.join(_SCHEMAS_DIR, 'raw.json')
+
+    with open(_JSON_NAME) as j:
+        sd_new = simplejson.load(j)
+    sd_new_dfc = {
+        df['field_name'] : {
+            ch['label']['en'].replace(',', '') :
+                ch['value'] for ch in df['choices']}
```

Comment:
sorry, dict comprehension , not generator expression. You can use dict([(key, value) for ... ]) instead